src/test/java/com/danielfrak/code/keycloak/providers/rest/LegacyProviderTest.java

```diff
@@ -218,15 +218,35 @@ void getUserByIdShouldThrowException() {
     }
 
     @Test
-    void removeFederationLinkWhenCredentialUpdates() {
+    void removeFederationLinkWhenCredentialUpdatesAndDeleteMigrationRecord() {
         var input = mock(CredentialInput.class);
+        final String username = "user";
+
+        when(legacyUserService.removeByIdentifier(username))
+                .thenReturn(true);
+        when(userModel.getUsername())
+                .thenReturn(username);
         when(userModel.getFederationLink())
                 .thenReturn("someId");
-
+        MultivaluedHashMap<String, String> config = new MultivaluedHashMap<>();
+        config.put(DELETE_MIGRATION_RECORD, List.of("true"));
+        when(model.getConfig()).thenReturn(config);
         assertFalse(legacyProvider.updateCredential(realmModel, userModel, input));
+        assertTrue(legacyUserService.removeByIdentifier(username));
```

Comment:
You seem to want to check if the `legacyUserService::removeByIdentifier` was called, but the way you do it won't work (it will literally just call the mocked method).

The proper way to check that this method has been called would be something like:
```java
verify(legacyUserService).removeByIdentifier(username);
```

To clean up the test a bit (having a cleaner separation of Arrange/Act/Assert) and prevent misunderstandings like this, I think we could also refactor the line above it like:
```java
boolean result = legacyProvider.updateCredential(realmModel, userModel, input);

assertFalse(result);
```